sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalogSuite.scala

```diff
@@ -679,6 +687,44 @@ abstract class SessionCatalogSuite extends AnalysisTest with Eventually {
     }
   }
 
+
+    test("SPARK-34528 - named explicitly field in struct of a view") {
```

Comment:
nti: `SPARK-34528: `

Reply:
Does this test pass before  #31368 and fail after it?

Reply:
I just deploy and check on a 3.1 branch and 3.0 branch (i can not test before) and I have also the bug. I just update the jira with it.
#31368 seems backportable to 3.1 easily so this version can be fix.
It's more complex for 3.0

Reply:
So #31368 has nothing to do with this PR? If so let's not mention it in the PR description, which confuses people.

Reply:
Your are right I just update the PR. I discover it testing the master branch after i see the path but it's not releted to #31368